src/js/ptdataset.js

```diff
@@ -190,4 +191,25 @@ export default class PTDataset {
       vehicleJourney => vehicleJourney.code,
     );
   }
+
+  /**
+   * Update raw vehicle journey realtime data into an existing VehicleJourney object,
+   * fetch each object indexed by their code for fast lookup and update that object
+   * @param  {Object} _vehicleJourneys Raw vehicle realtime data
+   */
+  updateVehicleJourneys(_vehicleJourneys) {
+    for (const [code, { realtime, cancelled }] of Object.entries(_vehicleJourneys)) {
+      // Convert time in seconds since noon minus 12h to Date object
+      for (const rtVehicle of Object.values(realtime)) {
+        rtVehicle.times = rtVehicle.times.map(time =>
+          TimeUtils.secondsToDateObject(time, this.referenceDate));
+      }
+
+      const vehicleJourney = this.vehicleJourneys[code];
+      if (vehicleJourney !== undefined) {
```

Comment:
`if (typeof vehicleJourney !== 'undefined')

https://stackoverflow.com/questions/3390396/how-to-check-for-undefined-in-javascript